py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -390,3 +391,41 @@ def index_select(
     set_layer_name(gather_layer, target, f"{name}_gather", source_ir)
 
     return gather_layer.get_output(0)
+
+
+def scatter(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    dim: int,
+    index: Union[TRTTensor, np.ndarray, torch.Tensor],
+    src: Union[TRTTensor, int, float],
+) -> TRTTensor:
+    input_shape = input.shape
+    index_shape = index.shape
+    index_shape_list = list(index_shape)
+    if index.dtype == trt.int64:
+        index = cast_trt_tensor(ctx, index, trt.int32, name + "_cast_index_tensor")
```

Comment:
If the index must be `trt.int32` and no other types (`trt.float32`, `trt.float16`, etc.) are acceptable, then it is fine to remove the `if` statement, as the `cast_trt_tensor` function will not insert a cast if the type is already `int32`, as here:

https://github.com/pytorch/TensorRT/blob/4b993f8ee30fd02b7ab9cff47114a0538562cf81/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L147

Reply:
It would be required for `trt.int64` cases, which will be the case in our test cases since torch requires int64 inputs.